### PR TITLE
Docs: Bring coding files up to style and add license

### PR DESCRIPTION
**Pull request by jwrober**

### CODE REVIEW

File: docs/Coding/hacking.rst [hunk 1]
```diff
@@ -848,44 +881,44 @@ following commonly applicable lists are maintained:
...
    information sent to it.
 
-(All following cases exist in game.all_connections.)
+All following cases exist in game.all_connections.
 
-#. (:code:`pconn->established == 0`) TCP connection has been made, but initial Freeciv21 packets have not yet
-   been negotiated (join_game etc). Exists in :code:`game.all_connections` only. Should not be sent any
-   information except directly as result of :code:`join_game` etc packets, or server shutdown, or connection
+#. :code:`pconn->established == 0`: TCP connection has been made, but initial Freeciv21 packets have not yet
+   been negotiated (join_game etc.). Exists in :code:`game.all_connections` only. Should not be sent any
```
Comment (daavko) [inline docs/Coding/hacking.rst:892]:
```suggestion
   been negotiated (:code:`join_game` etc.). Exists in :code:`game.all_connections` only. Should not be sent any
```

File: docs/Coding/hacking.rst [hunk 2]
```diff
@@ -848,44 +881,44 @@ following commonly applicable lists are maintained:
...
+   information except directly as result of :code:`join_game` etc. packets, or server shutdown, or connection
    close, etc.
 
-(All following cases exist in game.est_connections.)
+All following cases exist in game.est_connections.
```
Comment (daavko) [inline docs/Coding/hacking.rst:896]:
```suggestion
All following cases exist in :code:`game.est_connections`.
```

**Review by daavko — COMMENTED** [review]
Too many occurences to go comment on individually: double spaces after periods. We don't use them in any of the newly written documentation, but it looks like whoever was writing the hacking guide originally was fond of them. Not sure whether it's worth it to add this to the style guide, or to just keep it in mind when bringing docs up to style.

File: docs/Coding/fcdb.rst [hunk 3]
```diff
@@ -133,20 +138,21 @@ The return status of all of these functions should be one of:
...
-* return a value from the :code:`--Database` configuration file :code:`fcdb.option(type)`.
+* Return the client-specified username in function :code:`auth.get_username(conn)`.
+* Return the client IP address (string) in function :code:`auth.get_ipaddr(conn)`.
+* Tell the server (the MD5 hash of) the correct password to check against.
+* For this connection (usually to be called by function :code:`user_load()`).
+* Returns whether this succeeded in function :code:`auth.set_password(conn, password)`.
+* Return (the MD5 hash of) the password for this connection (as specified by the client in in function
+  :code:`user_save()`, or as previously set by the function :code:`set_password()`.
+* The function :code:`auth.get_password(conn)` runs.
+* Return a value from the :code:`--Database` configuration file in function :code:`fcdb.option(type)`.
```
[the remark this answers is not in the document]
Comment (jwrober) [reply docs/Coding/fcdb.rst:151]: Actually this whole section is a mess. Needs a rewrite by a knowledgeable person. I'll open another issue after the merge.

**Pull request merged**